### PR TITLE
Convert `DllImport` to `LibraryImport` for recyclebin, clipboard, and computerinfo cmdlets

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
# PR Summary

Convert usage of `DllImport` to `LibraryImport` for some cmdlets in Management module.  Reviewed usage and removed `SetLastError` where the code doesn't retrieve the last error.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Review by iSazonov — APPROVED** [review]

**Comment by xtqqczze** [comment]
Contributes to #18553.

File: src/Microsoft.PowerShell.Commands.Management/commands/management/Clipboard.cs [hunk 1]
```diff
@@ -157,46 +157,46 @@ public static void SetRtf(string plainText, string rtfText)
...
 
-        [DllImport("kernel32.dll")]
+        [LibraryImport("kernel32.dll")]
         [return: MarshalAs(UnmanagedType.Bool)]
-        private static extern bool GlobalUnlock(IntPtr hMem);
+        private static partial bool GlobalUnlock(IntPtr hMem);
 
-        [DllImport("kernel32.dll", ExactSpelling = true, EntryPoint = "RtlMoveMemory", SetLastError = true)]
-        private static extern void CopyMemory(IntPtr dest, IntPtr src, uint count);
+        [LibraryImport("kernel32.dll", EntryPoint = "RtlMoveMemory")]
```
Comment (daxian-dbw) [inline src/Microsoft.PowerShell.Commands.Management/commands/management/Clipboard.cs:173]: `LibraryImport` has the `SetLastError` property too, so maybe set that to `true` as well?

File: src/Microsoft.PowerShell.Commands.Management/commands/management/Clipboard.cs [hunk 2]
```diff
@@ -157,46 +157,46 @@ public static void SetRtf(string plainText, string rtfText)
...
+        private static partial void CopyMemory(IntPtr dest, IntPtr src, uint count);
 
-        [DllImport("user32.dll", SetLastError = false)]
+        [LibraryImport("user32.dll")]
```
Comment (daxian-dbw) [inline src/Microsoft.PowerShell.Commands.Management/commands/management/Clipboard.cs:176]: Same here. If the `SetLastError` is removed intentionally, then maybe explain why.

File: src/Microsoft.PowerShell.Commands.Management/commands/management/Clipboard.cs [hunk 3]
```diff
@@ -157,46 +157,46 @@ public static void SetRtf(string plainText, string rtfText)
...
+        private static partial IntPtr GetClipboardData(uint format);
 
-        [DllImport("user32.dll")]
-        private static extern IntPtr SetClipboardData(uint format, IntPtr data);
+        [LibraryImport("user32.dll")]
+        private static partial IntPtr SetClipboardData(uint format, IntPtr data);
 
-        [DllImport("user32.dll", SetLastError = true)]
-        private static extern uint RegisterClipboardFormat(string lpszFormat);
+        [LibraryImport("user32.dll", StringMarshalling = StringMarshalling.Utf16)]
```
Comment (daxian-dbw) [inline src/Microsoft.PowerShell.Commands.Management/commands/management/Clipboard.cs:198]: There was no `CharSet = CharSet.Unicode` setting before, so why `StringMarshalling.Utf16` is needed in this case?

Comment (xtqqczze) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/Clipboard.cs:198]: Marshalling of string is apparently not supported without explicit marshalling information. 

https://github.com/dotnet/runtime/blob/9e8c7a8f65ad86ee68d316de4e3a5deea72b7edd/docs/design/libraries/LibraryImportGenerator/Compatibility.md#string-marshalling

Comment (xtqqczze) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/Clipboard.cs:198]: When `DllImportAttribute.CharSet` is not set, it is up to the language compiler to determine which CharSet to use. C#, uses the [None](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.charset#system-runtime-interopservices-charset-none) charset by default, which has the same behavior as the [Ansi](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.charset#system-runtime-interopservices-charset-ansi) charset.

https://learn.microsoft.com/en-us/dotnet/standard/native-interop/charset

Comment (daxian-dbw) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/Clipboard.cs:198]: Thanks @xtqqczze. It's called out more clearly [here](https://github.com/dotnet/runtime/blob/9e8c7a8f65ad86ee68d316de4e3a5deea72b7edd/docs/design/libraries/LibraryImportGenerator/Compatibility.md#semantic-changes-compared-to-dllimportattribute):
> The P/Invoke source generator makes no assumptions about StringMarshalling if it is not explicitly set on LibraryImportAttribute. Marshalling of char or string requires explicitly specifying marshalling information.

Then another question is, how to select between utf8 and utf16? Should we always use utf16 on Windows and utf8 on Unix platforms?

File: src/Microsoft.PowerShell.Commands.Management/commands/management/Clipboard.cs (hunk 1 above)
Comment (iSazonov) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/Clipboard.cs:173]: It is not used in callsite.

File: src/Microsoft.PowerShell.Commands.Management/commands/management/Clipboard.cs (hunk 2 above)
Comment (iSazonov) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/Clipboard.cs:176]: It is not used in callsite.

File: src/Microsoft.PowerShell.Commands.Management/commands/management/Clipboard.cs (hunk 3 above)
Comment (iSazonov) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/Clipboard.cs:198]: This depends on target API. On Windows most of APIs use UTF16, but LDAP uses UTF8 as per the standard, also BER string is used sometimes (crypto API?).

**Pull request merged**